src/catalog/postgres/pg_proc_impl.cpp

```diff
@@ -0,0 +1,791 @@
+  std::vector<storage::TupleSlot> index_results;
+  procs_oid_index_->ScanKey(*txn, *key_pr, &index_results);
+  NOISEPAGE_ASSERT(
+      index_results.size() == 1,
+      "Incorrect number of results from index scan. Expect 1 because it's a unique index. 0 implies that function was "
+      "called with an oid that doesn't exist in the Catalog, which implies a programmer error. There's no reasonable "
+      "code path for this to be called on an oid that isn't present.");
+
+  auto *select_pr = pg_proc_ptr_pri_.InitializeRow(buffer);
+  const auto result UNUSED_ATTRIBUTE = procs_->Select(txn, index_results[0], select_pr);
+  NOISEPAGE_ASSERT(result, "Index already verified visibility. This shouldn't fail.");
+
+  auto *ptr_ptr = (reinterpret_cast<void **>(select_pr->AccessWithNullCheck(0)));
+
+  execution::functions::FunctionContext *ptr;
+  if (ptr_ptr == nullptr) {
+    ptr = nullptr;
+  } else {
+    ptr = *reinterpret_cast<execution::functions::FunctionContext **>(ptr_ptr);
+  }
+
+  delete[] buffer;
+  return common::ManagedPointer<execution::functions::FunctionContext>(ptr);
+}
+
+proc_oid_t PgProcImpl::GetProcOid(const common::ManagedPointer<DatabaseCatalog> dbc,
+                                  const common::ManagedPointer<transaction::TransactionContext> txn,
+                                  namespace_oid_t procns, const std::string &procname,
+                                  const std::vector<type_oid_t> &arg_types) {
+  auto name_pri = procs_name_index_->GetProjectedRowInitializer();
+  byte *const buffer = common::AllocationUtil::AllocateAligned(pg_proc_all_cols_pri_.ProjectedRowSize());
+
+  auto name_pr = name_pri.InitializeRow(buffer);
+  auto name_map = procs_name_index_->GetKeyOidToOffsetMap();
+
+  auto name_varlen = storage::StorageUtil::CreateVarlen(procname);
+  auto all_arg_types_varlen = storage::StorageUtil::CreateVarlen(arg_types);
+  *reinterpret_cast<namespace_oid_t *>(name_pr->AccessForceNotNull(name_map[indexkeycol_oid_t(1)])) = procns;
+  *reinterpret_cast<storage::VarlenEntry *>(name_pr->AccessForceNotNull(name_map[indexkeycol_oid_t(2)])) = name_varlen;
+
+  std::vector<storage::TupleSlot> results;
+  procs_name_index_->ScanKey(*txn, *name_pr, &results);
+
+  proc_oid_t ret = INVALID_PROC_OID;
+  std::vector<proc_oid_t> matching_functions;
+  if (!results.empty()) {
+    const std::vector<type_oid_t> variadic = {dbc->GetTypeOidForType(type::TypeId::VARIADIC)};
+    auto variadic_varlen = storage::StorageUtil::CreateVarlen(variadic);
+
+    // Search through results and see if any match the parsed function by argument types
+    for (auto &tuple : results) {
+      auto table_pr = pg_proc_all_cols_pri_.InitializeRow(buffer);
+      bool UNUSED_ATTRIBUTE visible = procs_->Select(txn, tuple, table_pr);
+      storage::VarlenEntry index_all_arg_types = *reinterpret_cast<storage::VarlenEntry *>(
+          table_pr->AccessForceNotNull(pg_proc_all_cols_prm_[postgres::PgProc::PROALLARGTYPES_COL_OID]));
```

Comment:
This is the wrong column to reference as this column includes out arguments which Postgres does *not* use to select functions (see [catalog documentation](https://www.postgresql.org/docs/12/catalog-pg-proc.html)).  It should be using `PROARGTYPES_COL_OID` to match Postgres' behavior.

Reply:
Fixed, good catch!